features/admin.webhooks.v1/pages/webhook-settings-page.tsx

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Box from "@oxygen-ui/react/Box";
+import Skeleton from "@oxygen-ui/react/Skeleton";
+import Typography from "@oxygen-ui/react/Typography";
+import { AppConstants } from "@wso2is/admin.core.v1/constants/app-constants";
+import { history } from "@wso2is/admin.core.v1/helpers/history";
+import { AlertLevels } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { Field, Form } from "@wso2is/form";
+import { RadioChild } from "@wso2is/forms";
+import {
+    EmphasizedSegment,
+    PageLayout } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import updateWebhookMetadata from "../api/update-webhook-metadata";
+import useGetWebhooksMetadata from "../api/use-get-webhooks-metadata";
+import { EventPublishingOrgSharePolicy } from "../models/shared-access";
+import {
+    WebhookMetadataUpdateRequestInterface,
+    WebhookSettingsFormPropsInterface,
+    WebhookSettingsFormValuesInterface
+} from "../models/webhook-settings";
+import { useHandleWebhookError } from "../utils/alert-utils";
+import "./webhook-settings-page.scss";
+
+const FORM_ID: string = "webhook-settings";
+
+/**
+ * This will be used to render the event publishing organization policy radio options.
+ */
+const ORGANIZATION_POLICY_RADIO_OPTIONS: RadioChild[] = [
+    {
+        label: "webhooks:pages.settings.organizationPolicy.radioOptions.currentOrgOnly",
+        value: EventPublishingOrgSharePolicy.NO_SHARING
+    },
+    {
+        label: "webhooks:pages.settings.organizationPolicy.radioOptions.currentOrgAndImmediateChild",
+        value: EventPublishingOrgSharePolicy.IMMEDIATE_EXISTING_AND_FUTURE_ORGS
+    }
+];
+
+/**
+ * Form to edit webhook settings.
+ *
+ * @param props - Props injected to the component.
+ * @returns Functional Component.
+ */
+export const WebhookSettingsForm: FunctionComponent<WebhookSettingsFormPropsInterface> = ({
+    ["data-componentid"]: componentId
```

Comment:
Let's use js default params instead of default props.

```suggestion
    ["data-componentid"]: componentId = "webhook-settings-page"
```

Reply:
This is update with d4ba4ce5f28de4329e3157e0b963ac69abe7e38e